src/main/scala/cromwell/engine/EngineFunctions.scala

```diff
@@ -0,0 +1,28 @@
+package cromwell.engine
+
+import cromwell.binding.IoInterface
+import cromwell.binding.expression.WdlStandardLibraryFunctions
+import cromwell.binding.values._
+
+import scala.language.postfixOps
+import scala.util.{Success, Try}
+
+/**
+ * Default implementation of Wdl Standard Library functions executed at the workflow level.
+ * This class is abstract to enforce that Backends extend it and customize the behavior if needed.
+ */
+abstract class WorkflowEngineFunctions(override val interface: IoInterface, val context: WorkflowContext) extends WdlStandardLibraryFunctions {
+
+  override def tempFilePath: String = context.root
+  override def globPath(glob: String): String = context.root
```

Comment:
So what happens is "how to glob" is independent from the backend, it only depends on the file system (Shared, GCS). This is good because it allows to have a `glob` method in `IoInterface`, that each file system implements to say how to glob some files out from a directory path.

However "where to glob from" is different depending on the backend. For instance local backend doesn't have anything specific regarding globbing, so you can just apply the pattern to the call directory. But JES delocalizes all the files into a dedicated directory inside the call directory. And this is where you need to go and look for the files.

There are different way to do it but in the end what's needed is a way for each backend to tell the `glob` function in `WdlStandardLibraryFunctions` where to glob from, and this the what this `globPath` method is for. It's only used in the `glob` function implementation in `WdlStandardLibraryFunctions`. And it's overridden in `JesWorkflowEngineFunctions`:

```
override def globPath(glob: String) = s"${context.root}/glob-${glob.md5Sum}/"
```

The more general idea is to minimize the need to reimplement engine functions for each backend, and to do that we need to identify the smaller set of things needed to implement them that are backend specific, so Backends only have those to implement.
The advantage of this is that if you look at 
`JesWorkflowEngineFunctions`, `JesCallEngineFunctions`, `LocalWorkflowEngineFunctions`and  `LocalCallEngineFunctions` they're almost empty, because everything is already in `WdlStandardLibraryFunctions`.